crawl-ref/source/webserver/util.py

```diff
@@ -132,13 +138,24 @@ def send_email(to_address, subject, body_plaintext, body_html):
         email_server.sendmail(config.smtp_from_addr, to_address, msg.as_string())
     finally:
         # end connection
-        if email_server: email_server.quit()
+        if connected: email_server.quit()
 
 def validate_email_address(address):  # type: (str) -> Optional[str]
-    # Returns an error string describing the problem, or None
+    # Returns an error string describing the problem, or None.
+    # NOTE: a blank email address validates successfully.
     if not address: return None
     if " " in address: return "Email address can't contain a space"
     if "@" not in address: return "Expected email address to contain the @ symbol"
     if not re.match(r"[^@]+@[^@]+\.[^@]+", address): return "Invalid email address"
     if len(address) >= 80: return "Email address can't be more than 80 characters"
     return None
+
+def humanise_bytes(num):
```

Comment:
this looks basically fine but here's an alternative version of this that I wrote for the scoring code which does the math slightly differently(it's 2-space indent sry):

```
def fmt_byte_size(b, precision=1):
  if b < 1000:
    return str(b) + 'B'
  b = b / 1000.0
  fmt = "%%.%df" % precision
  for suffix in ['KB', 'MB', 'GB', 'TB']:
    if int(b / 1000) == 0 or suffix == 'TB':
      return (fmt % b) + suffix
    b = b / 1000.0
```

If one were to be really pedantic, my code should probably use `kB` etc and yours should say `kibibytes` etc.